File: 0455-assign-cookies/0455-assign-cookies.py
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        if len(s) == 0:
            return 0

        g.sort()
        s.sort()

        cookie, child = 0,0
        res = 0

        while cookie < len(s) and child < len(g):
            if s[cookie] >= g[child]:
                res+=1
                child+=1
            
            cookie+=1
        

        return res

File: 0455-assign-cookies/0455-assign-cookies.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        int cookies = s.length;
        if (cookies == 0) return 0;

        Arrays.sort(g);
        Arrays.sort(s);

        int curr = 0;
        int content = 0;

        for (int i = 0; i < g.length; i++) {
            while (s[curr] < g[i]) {
                curr++;
                if (curr == cookies) return content;
            }

            content++;
            curr++;
            if (curr == cookies) return content;
        }

        return content;
    }
}
